File: app/src/test/java/com/hana053/micropost/testing/TestApplication.java
package com.hana053.micropost.testing;

import android.os.StrictMode;

import com.hana053.micropost.presentation.core.di.AppComponent;
import com.hana053.micropost.presentation.core.base.BaseApplication;
import com.hana053.micropost.presentation.core.di.DaggerAppComponent;
import com.hana053.micropost.system.SystemServicesModule;

import timber.log.Timber;

public class TestApplication extends BaseApplication {

    @Override
    public void onCreate() {
        super.onCreate();

        StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder().detectAll().penaltyLog().penaltyDeathOnNetwork().build());
        StrictMode.setVmPolicy(new StrictMode.VmPolicy.Builder().detectLeakedSqlLiteObjects().detectLeakedClosableObjects().detectActivityLeaks().penaltyLog().build());

        Timber.plant(new Timber.DebugTree());
    }

    @Override
    protected AppComponent createComponent() {
        return DaggerAppComponent.builder()
                .systemServicesModule(new SystemServicesModule(this))
                .build();
    }

    public AppComponent getComponent() {
        return component;
    }
}

File: app/src/test/java/com/hana053/micropost/testing/TestApplication.kt
package com.hana053.micropost.testing

import android.os.StrictMode
import android.support.annotation.VisibleForTesting
import com.github.salomonbrys.kodein.Kodein
import com.github.salomonbrys.kodein.lazy

import com.hana053.micropost.BaseApplication
import com.hana053.micropost.services.serviceModule
import com.hana053.micropost.system.systemServiceModule

import timber.log.Timber

class TestApplication : BaseApplication() {

    override fun onCreate() {
        super.onCreate()

        StrictMode.setThreadPolicy(StrictMode.ThreadPolicy.Builder().detectAll().penaltyLog().penaltyDeathOnNetwork().build())
        StrictMode.setVmPolicy(StrictMode.VmPolicy.Builder().detectLeakedSqlLiteObjects().detectLeakedClosableObjects().detectActivityLeaks().penaltyLog().build())

        Timber.plant(Timber.DebugTree())
    }

    fun  setKodein(kodein: Kodein) {
       _kodein = kodein
    }




}
